flake-module.nix

```diff
@@ -71,7 +71,7 @@ in
               };
             };
           };
-          projectSubmodule = types.submodule (args@{ name, config, lib, ... }:
+          projectSubmodule = types.submoduleWith { specialArgs = { inherit pkgs self; }; modules = [ ./haskell-project.nix (args@{ name, config, lib, ... }:
```

Comment:
All this happening on a single line is ugly, but I think almost all of the things that would otherwise be reindented below can move to the `haskell-project.nix` module later. Doing that here would cause a large diff (again).